zerver/views/upload.py

```diff
@@ -88,12 +93,16 @@ def serve_local(
     attachment = download or mimetype not in INLINE_MIME_TYPES
```

Comment:
Does `generate_unauthed_file_access_url` need to be patched with these changes? I believe the intent of that is to provide a URL that can be used by mobile client code paths that don't support automatically following redirects, and I think now that we're not serving a redirect in general, we probably no longer want to have that be a redirect to S3.

Reply:
I'm not sure how `generate_unauthed_file_access_url` is related to this commit or hunk?

If you mean that in general the URL whether the URL that `generate_unauthed_file_access_url` returns should be direct to S3 (as it used to be) or to the Zulip server, the final commit in this series takes the stance of making it to the same `/user_uploads/temporary/` endpoint as when the non-S3 backend is in use, and does the same proxying.  This makes access to the bucket (even temporary token-protected ones) be _only_ through the Zulip server, and presents a unified URL when switching between S3-backed and file-backed.

Reply:
Oh, huh, I saw this comment originally on the second-to-last-commit of the series -- but I _also_ see it on the last one.

Reply:
For reference, this decision is buried in the last paragraph of the last commit:

    The `Content-Disposition` header logic can now also be entirely shared
    with the local-file codepath, as can the `url_only` path used by
    mobile clients.  While we could provide the direct-to-S3 temporary
    signed URL to mobile clients, we choose to provide the
    served-from-Zulip signed URL, to better control caching headers on it,
    and greater consistency.  In doing so, we adjust the salt used for the
    URL; since these URLs are only valid for 60s, the effect of this salt
    change is minimal.

Reply:
> Oh, huh, I saw this comment originally on uploads: Provide our own Content-Disposition header. -- but I also see it on the one after that.

Yeah, GitHub's comments aren't tied to a commit, they'd tied to a place in the code. It's pretty weird but probably helps prevent them disappearing during rebases.